builder/softlayer/client.go

```diff
@@ -246,13 +246,20 @@ func (self SoftlayerClient) CreateInstance(instance InstanceType) (map[string]in
 		}
 	} else {
 		instanceRequest.OsReferenceCode = instance.BaseOsCode
-		instanceRequest.BlockDevices = []*BlockDevice{
-			&BlockDevice{
-				Device: "0",
+
+		instanceRequest.BlockDevices = []*BlockDevice{}
+
+		for index, element := range instance.DiskCapacities {
+			var realIndex interface{} = index // deviceIndex 0 is reserved for swap, need to workaround that
```

Comment:
It looks like I lost part of my alternate suggestion here. Rather than restating it, I'll await your clarification on how this is supposed to work. We will definitely want to revise how you get to that goal, whatever it may be.